src/main/scala/FSimManager.scala

```diff
@@ -0,0 +1,94 @@
+package testchipip
+
+import chisel3._
+import chisel3.core.IntParam
+import chisel3.util._
+import freechips.rocketchip.config.{Field, Parameters}
+import freechips.rocketchip.coreplex.CacheBlockBytes
+import freechips.rocketchip.coreplex.{HasSystemBus, HasPeripheryBus}
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.regmapper.{RegisterReadIO,RegisterWriteIO ,RegField, HasRegMap}
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.util.{ParameterizedBundle, DecoupledHelper, UIntIsOneOf}
+import scala.math.max
+
+class FSimManagerIO extends Bundle {
+  val req = Decoupled(UInt(32.W))
+  val resp = Flipped(Decoupled(UInt(32.W)))
+}
+
+case class FSimManagerControllerParams(address: BigInt, beatBytes: Int)
+
+trait FSimManagerControllerBundle extends Bundle {
+  val ext = new FSimManagerIO
+}
+
+trait FSimManagerControllerModule extends Module with HasRegMap {
+  val io: FSimManagerControllerBundle
+
+  //val query = Reg(Bool())
+  //val query = Reg(UInt(32.W))
+  val requestQueue = Module(new Queue(UInt(32.W),10))
+  val responseQueue = Module(new Queue(UInt(32.W),10))
+
+  //val data = Reg(UInt(32.W))
+  val allocRead = Wire(new RegisterReadIO(UInt(32.W)))
+  val requestReg = Reg(UInt(32.W))
+
+  io.ext.req <> requestQueue.io.deq
+  responseQueue.io.enq <> io.ext.resp
+ 
+  allocRead.request.ready := true.B
+
+  when(!(responseQueue.io.count > 0.U)) {
+    allocRead.response.valid := true.B
+    allocRead.response.bits := 0.U
+  } .otherwise {
+    allocRead.response <> responseQueue.io.deq
+  }
+
+  regmap(
+    0x00 -> Seq(RegField.w(32, requestQueue.io.enq)),
```

Comment:
As I and Sagar discussed before, this device might be generalized to control FireSim in the future.
So I made this device require 32-bit request value to be written, instead of just returning rtc value on read.

Reply:
Please comment if you still think it's the problem.